packages/devtools_app/lib/src/network/flutter/network_screen.dart

```diff
@@ -0,0 +1,312 @@
+    });
+  }
+
+  static const _headerTextStyle = TextStyle(
+    color: Colors.white,
+    fontSize: 14,
+    fontWeight: FontWeight.bold,
+  );
+
+  @override
+  Widget build(BuildContext context) {
+    networkController.initialize();
+    const double minIncludeTextWidth = 600;
+    return ValueListenableBuilder<HttpRequests>(
+      valueListenable: networkController.requestsNotifier,
+      builder: (context, data, widget) {
+        dataTableSource.data = data.requests;
+        return ValueListenableBuilder(
+          valueListenable: networkController.recordingNotifier,
+          builder: (context, recording, widget) => Column(
+            children: [
+              Row(
+                children: [
+                  recordButton(
+                    recording: recording,
+                    minIncludeTextWidth: minIncludeTextWidth,
+                    onPressed: () => networkController.startRecording(),
+                  ),
+                  pauseRecordingButton(
+                    recording: recording,
+                    minIncludeTextWidth: minIncludeTextWidth,
+                    onPressed: () => networkController.pauseRecording(),
+                  ),
+                  OutlineButton(
+                    onPressed: () => networkController.refreshRequests(),
+                    child: Label(
+                      FlutterIcons.refresh,
+                      'Refresh',
+                      minIncludeTextWidth: 900,
+                    ),
+                  ),
+                  const SizedBox(width: 8.0),
+                  clearButton(
+                    onPressed: () {
+                      dataTableSource.clearSelection();
+                      networkController.clear();
+                    },
+                  ),
+                  const Spacer()
+                ],
+              ),
+              ValueListenableBuilder(
+                valueListenable: dataTableSource.currentSelectionListenable,
+                builder: (context, HttpRequestData data, widget) => Expanded(
+                  child: (!recording && dataTableSource.rowCount == 0)
+                      ? Container(
+                          child: Center(
+                            child: recordingInfo(
+                              recording: recording,
+                              recordedObject: 'HTTP requests',
+                              isPause: true,
+                            ),
+                          ),
+                        )
+                      : Split(
+                          initialFirstFraction: 0.55,
+                          axis: Axis.horizontal,
+                          firstChild: (dataTableSource.rowCount == 0)
+                              ? Container(
+                                  child: const Center(
+                                      child: CircularProgressIndicator()),
+                                )
+                              : Scrollbar(
```

Comment:
Separate this out into a helper method or widget that takes in its data source dependencies and builds the table.